### DISCUSSION TITLE
Authentication (Registration and Login)

### DISCUSSION THREAD

**Discussion by milack68** [Webmasters-Europe/codemeetup-forum#34] [Ideas]
Aus Datenschutzgründen  und um Postings auch zuordnen zu können sollte es nicht möglich sein, dass nicht registrierte Personen Themen und Postings erstellen können.
Aus diesem Grund sollte eine Authentifizierung implementiert werden.

Ein User muss sich mit folgenden **Daten** registrieren
- Username
- Email-Adresse
- Passwort

Die **Email-Adresse soll verifiziert** werden, indem an die eingegeben Adresse ein Link geschickt wird, den der User dann bestätigen muss.
Das **Passwort soll nicht im Klartext in der Datenbank** gespeichert werden. Es soll eine **Passwort-Vergessen-Funktion** implementiert werden.

Nach Registrierung und Bestätigung der E-Mail-Adresse kann der User das Forum nutzen.

Der User soll sich alternativ mit seinem Usernamen oder mit seiner E-Mail-Adresse anmelden können.

**Comment by alenavershinina** [comment]
Ich finde Anmeldung mit oAuth (Google, Facebook oder Github account) viel einfacher fuer Benutzer und wir haben keine Probleme mit Passwort Schutz (hashing/encrypting)

**Comment by milack68** [comment]
> Ich finde Anmeldung mit oAuth (Google, Facebook oder Github account) viel einfacher fuer Benutzer und wir haben keine Probleme mit Passwort Schutz (hashing/encrypting)

Ich würde beides umsetzen - sowohl oAuth als auch die Möglichkeit sich selbst zu registrieren. Nicht jeder möchte oAuth nutzen (ich zB habe für alle geschützten Bereiche auf Websites eigene Accounts und jeweils ein eigenes Passwort) oder hat einen entsprechenden Account und der Aufwand für alle Möglichkeiten (Apple, Google, Facebook, Github, ...) oAuth anzubieten ist meiner Meinung nach größer als das Passwort Hashing. Das ist eigentlich nur ein einzelner Befehl in Laravel.

**Comment by milack68** [comment]
Ich schließe diesen Issue - wir haben bzw. werden die Anmeldung in einzelne Issues (zB oAuth with Google, login with username, ...) aufgeteilt bzw. aufteilen.

**Comment by Cosnavel** [comment]
@milack68 m.M. nach sollten wir den Inhalt des Issues in die Discussions verschieben. Oder du erstellst für Email Verifikation noch nen Issue.

**Reply by milack68** [reply]
Jetzt können wir über den Inhalt diskutieren.

**Reply by Cosnavel** [reply]
Habe nen Issue zur Email Verifikation erstellt

**Reply by Cosnavel** [reply]
#39
